packages/flutter/lib/src/widgets/radio_base.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+import 'ticker_provider.dart';
+import 'toggleable.dart';
+import 'widget_state.dart';
+
+/// Signature for [RadioBase.painterGetter]
+///
+/// The getter can use `state` to configure the painter before returning.
+typedef RadioPainterGetter = CustomPainter Function(ToggleableStateMixin state);
+
+/// A base class for Radio button that provides basic radio functionalities.
+///
+/// This widget uses painter returned from `painterGetter` to draw the radio.
+/// Consider using [ToggleablePainter] as a base class to implement the painter.
+///
+/// This widget allows selection between a number of mutually exclusive values.
+/// When one radio button in a group is selected, the other radio buttons in the
+/// group cease to be selected. The values are of type `T`, the type parameter
+/// of the radio class. Enums are commonly used for this purpose.
+///
+/// The radio button itself does not maintain any state. Instead, selecting the
+/// radio invokes the [onChanged] callback, passing [value] as a parameter. If
+/// [groupValue] and [value] match, this radio will be selected. Most widgets
+/// will respond to [onChanged] by calling [State.setState] to update the
+/// radio button's [groupValue].
+///
+/// See also:
+///
+///  * [Radio], which uses this widget to build a material styled radio button.
+///  * [CupertinoRadio], which uses this widget to build a iOS styled radio button.
+class RadioBase<T> extends StatefulWidget {
+  /// Creates a radio button.
+  ///
+  /// The radio button itself does not maintain any state. Instead, when the
+  /// radio button is selected, the widget calls the [onChanged] callback. Most
+  /// widgets that use a radio button will listen for the [onChanged] callback
+  /// and rebuild the radio button with a new [groupValue] to update the visual
+  /// appearance of the radio button.
+  ///
+  /// The all arguments except `key` are required:
```

Comment:
Consider cutting this comment, I think this is pretty self-evident by reading the parameters :)